modules/chat/client/controllers/chat.client.controller.js

```diff
@@ -1,43 +1,55 @@
-'use strict';
+(function () {
+  'use strict';
 
-// Create the 'chat' controller
-angular.module('chat').controller('ChatController', ['$scope', '$location', 'Authentication', 'Socket',
-  function ($scope, $location, Authentication, Socket) {
-    // Create a messages array
-    $scope.messages = [];
+  angular
+    .module('chat')
+    .controller('ChatController', ChatController);
 
-    // If user is not signed in then redirect back home
-    if (!Authentication.user) {
-      $location.path('/');
-    }
+  ChatController.$inject = ['$scope', '$state', 'Authentication', 'Socket'];
 
-    // Make sure the Socket is connected
-    if (!Socket.socket) {
-      Socket.connect();
-    }
+  function ChatController($scope, $state, Authentication, Socket) {
+    var vm = this;
```

Comment:
Would we rather not do "chat", i.e. more appropriate, less generic names? I think it's useful in the HTML and reads better, but it's up to popular opinion as it won't really affect anything other than variable naming within the controllers.